java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1853,21 +1833,35 @@ public static ColumnVector concatenate(ColumnVector... columns) {
     private final TimeUnit tsTimeUnit;
     private HostMemoryBuffer data;
     private HostMemoryBuffer valid;
+    private HostMemoryBuffer offsets;
     private long currentIndex = 0;
     private long nullCount;
+    private long stringBufferSize = 0;
+    private int currentStringByteIndex = 0;
     private boolean built;
 
     /**
      * Create a builder with a buffer of size rows
      * @param type       datatype
      * @param tsTimeUnit for TIMESTAMP the unit of time it is storing.
      * @param rows       number of rows to allocate.
+     * @param stringBufferSize the size of the string data buffer if we are
+     *                         working with Strings.  It is ignored otherwise.
      */
-    Builder(DType type, TimeUnit tsTimeUnit, long rows) {
+    Builder(DType type, TimeUnit tsTimeUnit, long rows, long stringBufferSize) {
       this.type = type;
       this.tsTimeUnit = tsTimeUnit;
       this.rows = rows;
-      this.data = HostMemoryBuffer.allocate(rows * type.sizeInBytes);
+      if (type == DType.STRING || type == DType.STRING_CATEGORY) {
+        this.data = HostMemoryBuffer.allocate(stringBufferSize);
```

Comment:
nit, we should probably assert if `stringBufferSize` is not at least `rows` bytes or something like that.

Reply:
That is not required, because if it is a row of nulls strings, and null string takes up no data.

Reply:
:+1: thanks.